core/domain/image_services_test.py

```diff
@@ -0,0 +1,128 @@
+            ValueError, r'Scaling factor should be in the interval \(0, 1].')
+        with value_exception:
+            image_services.compress_image(self.jpeg_raw_image, 1.1)
+
+    def test_compression_results_in_correct_format(self):
+        compressed_image = (
+            image_services.compress_image(self.jpeg_raw_image, 0.7))
+        pil_image = Image.open(io.BytesIO(compressed_image))
+        self.assertEqual(pil_image.format, 'JPEG')
+
+        compressed_image = (
+            image_services.compress_image(self.png_raw_image, 0.7))
+        pil_image = Image.open(io.BytesIO(compressed_image))
+        self.assertEqual(pil_image.format, 'PNG')
+
+    def test_compression_results_in_identical_files(self):
+        def is_identical(file1, file2):
+            """Checks the differences between the two images and returns if they
+            are different.
+
+            Args:
+                file1: str. Content of first image.
+                file2: str. Content of second image.
+
+            Returns:
+                bool. Returns whether the images are different.
+            """
+            image1 = Image.open(io.BytesIO(file1)).convert('RGB')
+            image2 = Image.open(io.BytesIO(file2)).convert('RGB')
+            diff = ImageChops.difference(image1, image2)
+
+            # Function diff.getbbox() returns a bounding box on all islands or
+            # regions of non-zero pixels. In other words, if we have a bounding
+            # box, there will be areas that are not 0 in the difference meaning
+            # that the 2 images are not equal.
+            return not diff.getbbox()
+
+        with python_utils.open_file(
+            os.path.join(
+                feconf.TESTS_DATA_DIR, 'compressed_image.jpg'),
+            mode='rb', encoding=None) as f:
+            correct_compressed_image = f.read()
+        correct_height, correct_width = (
+            image_services.get_image_dimensions(correct_compressed_image))
+        compressed_image = (
+            image_services.compress_image(self.jpeg_raw_image, 0.5))
+
+        # In order to make sure the images are the same, the function needs to
+        # open and save the image specifically using PIL since the "golden
+        # image" (image that we compare the compressed image to) is saved using
+        # PIL. This applies a slight quality change that won't appear unless we
+        # save it using PIL.
+        temp_image = Image.open(io.BytesIO(compressed_image))
+        image_format = temp_image.format
+        with io.BytesIO() as output:
+            temp_image.save(output, format=image_format)
+            compressed_image_content = output.getvalue()
+        height, width = image_services.get_image_dimensions(
+            compressed_image_content)
+        self.assertEqual(correct_height, height)
+        self.assertEqual(correct_width, width)
+        self.assertTrue(
+            is_identical(compressed_image_content, correct_compressed_image))
```

Comment:
I don't think you need a function for this. Just do it inline -- you only call that function once anyway.

Reply:
Done